packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -205,7 +220,11 @@ export class IndexedDbPersistence implements Persistence {
         .next(canActAsPrimary => {
           if (canActAsPrimary !== this.isPrimary) {
             this.isPrimary = canActAsPrimary;
-            this.queue.enqueue(() => this.primaryStateListener(this.isPrimary));
+            this.queue.enqueue(async () => {
+              if (this.started) {
```

Comment:
It's non-obvious why this check is necessary. Maybe it's guarding against early shutdown? Can you add a comment?

// Ensure persistence hasn't been shut down before notifying the listener.

Reply:
Comment added. This frequently happened during test failures, where the client is being shutdown, while another (unexpected) operation is still doing some leftover work.